File: myblogsbackend/src/main/java/com/ez/myblogsbackend/payload/PostSearchResponse.java
package com.ez.myblogsbackend.payload;

import java.util.Date;

public interface PostSearchResponse {

    // "postid" column
    public Long getPostid();

    // "categoryid" column
    public Long getCategoryid();

    // "categoryName" column
    public String getCategoryName();

    // "title" column
    public String getTitle();

    // "content" column
    public String getContent();

    // "createDatetime" column
    public Date getCreateDatetime();

    // "creatorid" column
    public Long getCreatorid();

    // "creatorEmail" column
    public String getCreatorEmail();

    // "postStatusid" column
    public long getPostStatusid();

    // "postStatusName" column
    public String getPostStatusName();

}

File: myblogsfrontend/src/app/payload/PostSearchResponse.ts
// list of posts by category id
export class PostSearchResponse {

    constructor(

        public postid: number,
        public categoryid: number,
        public categoryName: string,
        public title: string,
        public content: string,
        public createDatetime: Date,
        public creatorid: number,
        public creatorEmail: string,
        public postStatusid: number,
        public postStatusName: string

    ) {

    }

}
